src/pathfinding.ts

```diff
@@ -0,0 +1,173 @@
+import * as jkstra from "jkstra";
+import { GameState } from "./gamestate";
+
+interface HexBoardGraph {
+  graph: jkstra.Graph;
+  vertexMap: Map<string, jkstra.Vertex>;
+}
+
+interface Coordinates {
+  x: number;
+  y: number;
+}
+
+export function playerHasWon(gameState: GameState): boolean {
+  const hexBoardGraph = createGraphFromGameState(gameState);
+  return existsPathForPlayer(hexBoardGraph);
+}
+
+function createGraphFromGameState(gameState: GameState): HexBoardGraph {
+  let hexBoardGraph: HexBoardGraph = {
+    graph: new jkstra.Graph(),
+    vertexMap: new Map<string, jkstra.Vertex>(),
+  };
+
+  createVerticesFromBoard(gameState, hexBoardGraph);
+  createEdgesFromGameState(gameState, hexBoardGraph);
+
+  return hexBoardGraph;
+}
+
+function createVerticesFromBoard(
+  gameState: GameState,
+  hexBoardGraph: HexBoardGraph
+) {
+  // Add start and end nodes
+  addVertex(hexBoardGraph, "black-start");
+  addVertex(hexBoardGraph, "black-end");
+  addVertex(hexBoardGraph, "white-start");
+  addVertex(hexBoardGraph, "white-end");
+
+  // Add node for each cell on the board
+  gameState.board.forEach((row, y) => {
+    row.forEach((_cell, x) => {
+      addVertex(hexBoardGraph, `${y}-${x}`);
+    });
+  });
+}
+
+function createEdgesFromGameState(
+  gameState: GameState,
+  hexBoardGraph: HexBoardGraph
+) {
+  const stone_color = "black";
```

Comment:
Actually it's a choice to have it inside the function and not as a constant, since later it will have to be passed as one of the functions arguments.

Will fix the case though.